plugins/hypervisors/vmware/src/main/java/com/cloud/storage/resource/VmwareStorageProcessor.java

```diff
@@ -2684,6 +2701,30 @@ private boolean verifyAllHostsMountedDatastore(List<Pair<ManagedObjectReference,
         return lstHosts.size() == numHostsChecked;
     }
 
+    private boolean verifyAllHostsMountedDatastore2(List<HostMO> lstHosts, DatastoreMO dsMO) throws Exception {
+        int numHostsChecked = 0;
+
+        for (HostMO hostToMatchMO : lstHosts) {
+            List<DatastoreHostMount> datastoreHostMounts = dsMO.getHostMounts();
+
+            for (DatastoreHostMount datastoreHostMount : datastoreHostMounts) {
+                ManagedObjectReference morHost = datastoreHostMount.getKey();
+                HostMO hostMO = new HostMO(dsMO.getContext(), morHost);
+
+                if (hostMO.getHostName().equals(hostToMatchMO.getHostName())) {
+                    if (datastoreHostMount.getMountInfo().isMounted() && datastoreHostMount.getMountInfo().isAccessible()) {
+                        numHostsChecked++;
+                    }
+                    else {
```

Comment:
What about a log here to show the admin when a data storage is either not mounted or not accessible?

Reply:
@rafaelweingartner We could do that, but it's not technically a problem at this point in execution. For example, when we create a datastore on an iSCSI target, it can take up to 30 seconds or so for each host in the cluster to mount the new VMFS volume. When this code returns false, it is just telling the calling code that the VMFS volume is not mounted on all hosts in the cluster. The calling code will wait a bit longer and then re-invoke this method to see if the VMFS volume is now mounted on all hosts in the cluster. At two minutes of trying and failing, the calling code stops invoking this method and just fails the operation.

Reply:
I know it is not a problem, but it felt like a good thing to log. 
I mean, I did not check others methods that use this block of code to see if they would log meaningful messages in case of a false return here. Reading your explanation it seems that when the operation fails there is some sort of logging to help administrators to understand the problem.  If that is the case, I agree that we do not need to add anything here.